codec/enforceproto/extraneous_fields.go

```diff
@@ -0,0 +1,355 @@
+package enforceproto
```

Comment:
We can bike shed on this, and can always change it even after merge, I am open to changing it to whatever. I chose that name because the goal here is to enforce certain properties against proto messages so not only extraneous fields but mismatching types and later on other checks could be added. `typecheckproto` or `protovalidate` or `validateproto` could perhaps work, but the initial code logic is the harder and most important part :)

Reply:
Yep, we can always change it later. How about `unknownproto`?

Reply:
Sure, I've renamed it to unknownproto.